### PR TITLE
Helm chart verification

### PR DESCRIPTION
**Pull request by bryanl**
Fixing regressions in the ksonnet helm integration:

* handle chart versions like `v0.1.3` when finding the latest version
* make kubernetes cluster version available to charts
* add ks helm repository verification tool

### CODE REVIEW

File: hacks/ks-verify-helm.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+require 'timeout'
+require 'tmpdir'
+require 'uri'
+require 'yaml'
+
+DEFAULT_HELM_REPO_URL="https://kubernetes-charts.storage.googleapis.com/index.yaml"
+
+helm_repo_url = ARGV[0] || DEFAULT_HELM_REPO_URL
+
+tmp_dir = Dir.mktmpdir("ks-verify-helm-", "/tmp")
```
Comment (shomron) [inline hacks/ks-verify-helm.rb]: You might use `tmp_dir = Dir.mktmpdir("ks-verify-helm-", Dir.tmpdir)`

`Dir.tmpdir` on my macOS does not resolve to `/tmp`.

File: hacks/ks-verify-helm.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env ruby
```
Comment (shomron) [inline hacks/ks-verify-helm.rb]: Might be worthwhile to add a `.ruby-version` or a `Dockerfile` pointing to a supported ruby version.

File: pkg/helm/renderer.go [hunk 3]
```diff
@@ -167,7 +179,19 @@ func (r *Renderer) renderWithHelm(componentName, raw, chartPath string) (map[str
...
-	vals, err := chartutil.ToRenderValuesCaps(c, config, options, nil)
+
+	kubeVersion, err := r.k8sVersion()
+	if err != nil {
+		return nil, errors.Wrap(err, "setting Kubernetes version for Helm")
+	}
+
+	caps := &chartutil.Capabilities{
+		KubeVersion: &version.Info{
+			GitVersion: kubeVersion,
```
Comment (shomron) [inline pkg/helm/renderer.go]: Where does this ultimately get consumed? Are any of the other version-related fields in `version.Info` needed?

File: hacks/ks-verify-helm.rb (hunk 1 above)
Comment (bryanl) [reply hacks/ks-verify-helm.rb]: It does, but I should use `Dir.tmpdir` if I ever wanted this to work in windows.

File: hacks/ks-verify-helm.rb (hunk 2 above)
Comment (bryanl) [reply hacks/ks-verify-helm.rb]: It will work with any Ruby over 2.0, but the Dockefile idea is interesting.

File: hacks/ks-verify-helm.rb (hunk 1 above)
Comment (shomron) [reply hacks/ks-verify-helm.rb]: Mine resolves to something like this:

```
/var/folders/gl/9th4vs815bv_bwgkxs9rmgr40000gn/T/ks-verify-helm-20180710-11142-19aege7
```

File: pkg/helm/renderer.go (hunk 3 above)
Comment (bryanl) [reply pkg/helm/renderer.go]: It is consumed in the helm chart templates. None of the other fields are consumed within any charts in the stable repo.

Comment (bryanl) [reply pkg/helm/renderer.go]: Here's an example: https://github.com/kubernetes/charts/blob/74a23f7efbb332ea873df89f09d28654c48f6615/stable/cockroachdb/templates/_helpers.tpl#L5

**Pull request merged**